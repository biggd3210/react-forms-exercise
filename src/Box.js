import React from 'react';


function Box ({id, color, height, width, handleRemove}) {
    const remove = () => handleRemove(id);
    return (
        <div>
        <div
        style={{backgroundColor: color, height: parseInt(height), width: parseInt(width)}}
        />
        <button onClick={remove}>Remove The Box!</button>
        </div>
    )
}

export default Box;