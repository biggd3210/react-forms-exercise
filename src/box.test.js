import { render, screen } from '@testing-library/react';
import Box from './Box';

it("should render without crashing", () => {
    render(<Box color="black" height="20" width="25"/>);
});

it("should match the snapshot", () => {
    const { asFragment } = render(<Box color="black" height="20" width="25"/>);
    expect(asFragment()).toMatchSnapshot();
})