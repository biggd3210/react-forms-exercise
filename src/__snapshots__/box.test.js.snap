// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="background-color: black; height: 20px; width: 25px;"
    />
    <button>
      Remove The Box!
    </button>
  </div>
</DocumentFragment>
`;
